### ISSUE TITLE
SSH key doesn't get deleted

### ISSUE THREAD

**Issue by tarelda** [go-gitea/gitea#4348]
- Gitea version (or commit ref): 6efdcae
- Git version: Not relevant I think
- Operating system: gitea in docker
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ x ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [ x ] Not relevant
- Log gist:

## Description

First, I added key as a rw deploy key to repo. Pushing didn't do anything, so I re-added it as account key only to encounter this error: 

```Gitea: Invalid key ID
Invalid key ID[key-5]: public key does not exist [id: 5]
fatal: Could not read from remote repository.
```

Recreating repo didn't help.

**Comment by techknowlogick** [comment]
Are you able to select the "rewrite your authorized key file" from the admin page, and try again?

**Comment by tarelda** [comment]
@techknowlogick Small update with new scenario probably related with previous. 
Tried pull without key installed first. Pull obviously failed, so I added key. Still not worked so regenerated authorized_keys, but without luck. Readded key and repeated previous steps still with no success. I had to manually delete reference to key with lower id from authorized_keys and was finally able to pull sources.

**Comment by lunny** [comment]
Related #1357 and should be fixed by #5671. I think you could rewrite public keys on Admin UI to manually fix that.
